### ISSUE TITLE
Support `minorVersion` in `releaseBranches` to release patches for older major versions

### ISSUE THREAD

**Issue by jumic** [projen/projen#2813]
Currently, you can specify multiple release branches in `.projenrc`. For each release branch, you can define a major version that is developed on the specific release branch. 

For example, if you are developing v2, you can use projen to release new versions on the v1 branch:
- v1.1.0
- v1.1.1
- v1.2.0

However, it is not possible to release version v1.1.2 after v1.2.0 has been released.

In this feature request, I would like to allow releasing new versions on older major versions/branches. Therefore, I would like to add a new optional attribute `minorVersion` to `releaseBranches`, e.g. v1.0 in this example.

```ts
releaseBranches: {
  "v1": {
    majorVersion: 1,
  },
  "v1.0": {
    majorVersion: 1,
    minorVersion: 0,  // new
  },
},
```

Does this fit into the concept of projen? If so, I would probably implement this feature.

**Comment by mrgrain** [comment]
Yes that would be fine. Word of warning though: The code around releases is currently quite messy.
